### DISCUSSION TITLE
Is it possible to automate adding collections names as tags for items within them?

### DISCUSSION THREAD

**Discussion by al9alal** [windingwind/zotero-actions-tags#216] [Ideas]
Hope you are well 
Thanks for the great plugin and custom scripts. 
Based on Zotero Discussion:
[](https://forums.zotero.org/discussion/107747/plugin-actions-tags#latest)

Is  it  possible to create a script to:
A- right click an item and show the collections it resides into:,and then add these collections names as tags for the item.
b-Being able to apply this to multiple items at once? [To generate tags for all items existing already within collections in the library]
c-Can this possibly be automated for any new items to automatically add tags with the name of the collection they are directed to [there is currently a good plugin for this specific purpose but it has some issues and I think it was not updated to Zotero 7 yet]

I believe this script will be of great benefit for many users as it will relate collections more to tags and decrease some deficiencies of collections. 

Thanks again for all your great efforts in further developing Zotero and its related plug-ins.

**Comment by chliuhq** [comment]
https://github.com/Zar-rok/Zotero-Add-Collection-Tag
This plugin does not work with Zotero 7.0.

**Comment by garth74** [comment]
I like this idea. I use nested collections as way of creating conceptual hierarchies, but including collection membership as tags would be really nice for other applications.

However, there isn't a trigger event for adding an item to a collection. So, automatic updates would have to be at startup or close. In some cases it may work with the create item event but only when it's being created in the collection. 

Creating a script for setting the collection name as a tag wouldn't be difficult though. I'll mess with this later today and get back to you.

**Comment by garth74** [comment]
I've added a [script to a separate discussion for this](https://github.com/windingwind/zotero-actions-tags/discussions/245#discussion-6119937). It solves my needs, but I am happy to adjust it so others find it useful too.
